adoc/chapters/programming_interface.adoc

```diff
@@ -20853,6 +21520,13 @@ to represent the following address spaces:
 * [code]#access::address_space::private_space#
 * [code]#access::address_space::generic_space#
 
+The descriptions in this section use the type name [code]#+__swizzle__+# to
```

Comment:
In the case that these functions take a `__swizzle__` I don't think we specify whether the return type should also be a `__swizzle__` or if it should be a `vec`, I think in some cases the described operation may result in a `vec` but I'm not sure if that will always be the case.

If these functions are permitted to return a `__swizzle__`, should it be `__writeable_swizzle__` or `__const_swizzle__`?

Reply:
None of these builtin functions ever return a swizzled vec object.  The normal wording for the return type is usually:

>The return type is `NonScalar` unless `NonScalar` is the `__swizzle__` type, in which case the return type is
the corresponding `vec`.